lib/xcodeproj/project.rb

```diff
@@ -216,7 +216,11 @@ def initialize_from_file
       if object_version.to_i > Constants::LAST_KNOWN_OBJECT_VERSION
         raise '[Xcodeproj] Unknown object version.'
       end
-      root_object.product_ref_group = root_object.main_group['Products'] || root_object.main_group.new_group('Products')
+
+      # Projects can have product_ref_groups that are not listed in the main_groups["Products"]
+      if root_object.product_ref_group.nil?
+        root_object.product_ref_group = root_object.main_group['Products'] || root_object.main_group.new_group('Products')
```

Comment:
That works as well, it's a bit more implicit, but I don't really have an issue either way.
I'll push a change.